crates/optimism/rpc/src/historical.rs

```diff
@@ -171,6 +266,17 @@ impl<P> HistoricalRpcInner<P>
 where
     P: BlockReaderIdExt + TransactionsProvider + Send + Sync + Clone,
 {
+    /// Checks if a request should be forwarded to the historical endpoint (synchronous check).
+    fn should_forward_request(&self, req: &Request<'_>) -> bool {
+        match req.method_name() {
+            "debug_traceTransaction" |
+            "eth_getTransactionByHash" |
+            "eth_getTransactionReceipt" |
+            "eth_getRawTransactionByHash" => self.should_forward_transaction(req),
+            method => self.should_forward_block_request(method, req),
+        }
+    }
```

Comment:
this is the same as maybe_forward_request

can we reuse the `should_forward` checks in maybe_forward_request